terraform/modules/prom-ec2/prometheus/cloud.conf

```diff
@@ -33,19 +33,35 @@ write_files:
         # if alerts bucket exists then sync it, otherwise this cron runs but has no effect
         * * * * * root [ "${alerts_bucket}" != "" ] && aws s3 sync s3://${alerts_bucket}/prometheus/alerts/ /etc/prometheus/alerts --region=${region} --delete
   - content: |
-       #!/bin/bash
-       if file -s /dev/nvme1n1 | grep -q "/dev/nvme1n1: data"; then
-         mkfs -t 'ext4' -L 'prometheus_disk' '/dev/nvme1n1'
-       else
-         echo "disk already formatted"
-       fi
-    path: /root/format_disk.sh
-    permissions: 0755
-  - content: |
-       #!/bin/bash
-       UUID=$(blkid /dev/nvme1n1 -s UUID -o value)
-       echo "UUID=$UUID /mnt ext4 defaults,nofail 0 2" >> /etc/fstab
-    path: /root/generate_fstab.sh
+        echo 'Configuring prometheus EBS'
+        vol=""
+        while [ -z "$vol" ]; do
+          # adapted from
+          # https://medium.com/@moonape1226/mount-aws-ebs-on-ec2-automatically-with-cloud-init-e5e837e5438a
+          # [Last accessed on 2020-04-02]
+          vol=$(lsblk | grep -e disk | awk '{sub("G","",$4)} {if ($4+0 == ${data_volume_size}) print $1}')
```

Comment:
what's the purpose of the `+0`? cast to integer or something?

Reply:
Yes. It was used in the example I copied from and couldn't find a decent description in the awk documentation. As usual, though. stackoverflow helped:
https://stackoverflow.com/questions/5808971/casting-to-int-in-awk